config-svc/pkg/prometheus/config_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file  except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the  License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+const testYaml = `global:
```

Comment:
What happens with invalid YAML?

Reply:
Or valid YAML but invalid configuration?

Reply:
Invalid YAML: `yaml.Unmarshal` will return an error

Valid YAML but invalid configuration: the only part we care about is each managed `ScrapeConfig`, which gets unmarshaled with `yaml.Unmarshal`, which'll return an error.

Will add some tests to confirm this.

Reply:
Actually, tell a lie: if we get an invalid config, we'll just ignore it and spit it right back out, because what might be invalid to us may be valid to Prometheus. The only time this will error is when a "CMOS Managed" scrape_config is invalid - will still add a test to verify.

Reply:
OK cool, yeah we want to make sure we cope with any future changes to the config format that are not directly relevant.

Reply:
And tests will help if someone comes along and accidentally breaks something.